src/plugins/inputs/ban.js

```diff
@@ -0,0 +1,44 @@
+"use strict";
+
+var Chan = require("../../models/chan");
+var Msg = require("../../models/msg");
+
+exports.commands = [
+	"ban",
+	"unban",
+	"banlist"
+];
+
+exports.input = function(network, chan, cmd, args) {
+	if (chan.type !== Chan.Type.CHANNEL) {
+		chan.pushMessage(this, new Msg({
+			type: Msg.Type.ERROR,
+			text: `${cmd} command can only be used in channels.`
+		}));
+
+		return;
+	}
+
+	if (cmd !== "banlist" && args.length === 0) {
+		if (args.length === 0) {
+			chan.pushMessage(this, new Msg({
+				type: Msg.Type.ERROR,
+				text: `Usage: /${cmd} <nick>`
+			}));
+
+			return;
+		}
+	}
+
+	switch (cmd) {
+	case "ban":
+		network.irc.ban(chan.name, args[0]);
```

Comment:
I mean, this isn't any new functionality, because if we send a "/mode +b username" then it's exactly the same. All this does is basically symlink "/ban" to "/mode +b" with some checks that you are sending something. 

[The spec is vague on it.](https://tools.ietf.org/html/rfc1459#section-4.2.3)